File: app/src/main/java/com/kaciula/archiman/data/util/gson/ApiGsonAdapterFactory.java
package com.kaciula.archiman.data.util.gson;

import com.google.gson.TypeAdapterFactory;
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory;

@GsonTypeAdapterFactory
public abstract class ApiGsonAdapterFactory implements TypeAdapterFactory {

  public static ApiGsonAdapterFactory create() {
    return new AutoValueGson_ApiGsonAdapterFactory();
  }
}

File: app/src/main/java/com/kaciula/archiman/data/util/gson/ApiGsonAdapterFactory.kt
package com.kaciula.archiman.data.util.gson

import com.google.gson.TypeAdapterFactory
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory

@GsonTypeAdapterFactory
abstract class ApiGsonAdapterFactory : TypeAdapterFactory {
    companion object {

        fun create(): ApiGsonAdapterFactory {
            return AutoValueGson_ApiGsonAdapterFactory()
        }
    }
}